api.py

```diff
@@ -0,0 +1,75 @@
+from functools import lru_cache, partial
+from typing import Callable, Dict, List, Optional, Tuple
+import os
+
+import flask
+
+app = flask.Flask(__name__)
+
+FILESYSTEM_ROOT = os.path.abspath(os.sep)
+DATASETS_ROOT = os.path.join(FILESYSTEM_ROOT, 'data')
+
+
+def readable(dataset_name: str) -> str:
+    """Convert name on disk to readable for the user"""
+    return dataset_name.replace('_', ' ')
+
+
+def get_datasets() -> List[Dict[str, str]]:
+    """"Get datasets available in the store"""
+    datasets = filter(os.path.isdir, os.listdir(DATASETS_ROOT))
+    return [{"name": readable(name), "value": name} for name in datasets]
+
+
+def substitute_tags(tag_map: Dict[str, str], text: str) -> str:
+    """Substitute tags from the text for corresponding values in the map"""
+    for tag in tag_map:
+        text = text.replace('"' + tag + '"', tag_map[tag])
+    return text
+
+
+Substitutor = Optional[Callable[[str], str]]
+
+
+def datasets_substitutor() -> Substitutor:
+    """Factory of datasets substitutor"""
+    datasets = get_datasets()
+    parsed = str(datasets).replace("'", '"')
+    return partial(substitute_tags, {'$DATASETS': parsed})
```

Comment:
Just wondering, why `str(datasets).replace(apos, quote)` instead of `json.dumps(datasets)`?

Reply:
Sometimes simplest solutions are hiding somewhere... :D Accepted.